glotaran/model/dataset.py

```diff
@@ -1,42 +1,51 @@
 """Glotaran Dataset"""
 
-from typing import Tuple
 from copy import copy
+from typing import Callable, List, Tuple, Union
+import warnings
+
 import numpy as np
 
 
+class DimensionalityError(Exception):
+    """
+    Custom exception if data have the wrong dimensionality
+    """
+    pass
+
+
 class Dataset(object):
```

Comment:
This class have undergone alot of changes. you will have to adapt.

Reply:
Guess by that you mean a rebase after PR #150 was merged?